plaso/engine/artifacts_trie.py

```diff
@@ -0,0 +1,304 @@
+    if path == path_separator:
+      matching_artifacts = set()
+      if sub_root_node.artifacts_names:
+        matching_artifacts.update(sub_root_node.artifacts_names)
+      return list(matching_artifacts)
+    path_segments = path.strip(path_separator).split(path_separator)
+    matching_artifacts = set()
+    # Update self.artifacts_paths before starting the search.
+    self.artifacts_paths = self._GetArtifactsPaths(sub_root_node)
+
+    self._SearchTrie(
+      sub_root_node, '', path_segments, path_separator, matching_artifacts)
+    return list(matching_artifacts)
+
+  def _SearchTrie(
+      self,node, current_path, segments, path_separator, matching_artifacts):
+    """Searches the trie for paths matching the given path segments.
+
+    Args:
+      node (TrieNode): current trie node being traversed.
+      current_path (str): path represented by the current node.
+      segments (list[str]): remaining path segments to match.
+      path_separator (str): path separator.
+      matching_artifacts (set[str]): Set to store matching artifact names.
+    """
+    if node.artifacts_names:
+      for artifact_name in node.artifacts_names:
+        for artifact_path in self.artifacts_paths.get(artifact_name, []):
+          if self._ComparePathIfSanitized(
+              current_path, path_separator, artifact_path,
+              node.path_separator):
+            matching_artifacts.add(artifact_name)
+          elif glob.has_magic(artifact_path):
+            if self._MatchesGlobPattern(
+                artifact_path, current_path, node.path_separator):
+              matching_artifacts.add(artifact_name)
+
+    if not segments:
+      return
+
+    segment = segments[0]
+    remaining_segments = segments[1:]
+
+    # Handle glob characters in the current segment.
+    for child_segment, child_node in node.children.items():
+      # Compare the sanitized version of the path segment stored in
+      # the tree to the path segment from to the tool output as it
+      # sanitizes path segments before writing data to disk.
+      sanitized_child_segment = path_helper.PathHelper.SanitizePathSegments(
+              [child_segment]).pop()
+      if segment in (child_segment, sanitized_child_segment):
+        # If the child is an exact match, continue traversal.
+        self._SearchTrie(child_node, self._CustomPathJoin(
+            path_separator,
+            current_path, child_segment), remaining_segments, path_separator,
+            matching_artifacts)
+      # If the child is a glob, see if it matches.
+      elif glob.has_magic(child_segment):
+        if self._MatchesGlobPattern(
+                child_segment, segment, child_node.path_separator):
+          self._SearchTrie(child_node, self._CustomPathJoin(
+              path_separator,
+              current_path, segment), remaining_segments, path_separator,
+              matching_artifacts)
+          self._SearchTrie(
+              node,
+              self._CustomPathJoin(
```

Comment:
is there a specific reason to execute ` self._CustomPathJoin(path_separator, current_path, segment)` twice here? or could the result be stored in a local variable ?

Reply:
Note that I changed the code, but I'm not sure if there was a reason or not.

Reply:
this should work, thanks